src/main/java/crawlercommons/sitemaps/SiteMapParser.java

```diff
@@ -228,41 +214,28 @@ public AbstractSiteMap parseSiteMap(byte[] content, URL url) throws UnknownForma
      *             {@link java.net.URL}
      */
     public AbstractSiteMap parseSiteMap(String contentType, byte[] content, URL url) throws UnknownFormatException, IOException {
-        MediaType mediaType = MediaType.parse(contentType);
-
-        // Octet-stream is the father of all binary types
-        while (mediaType != null && !mediaType.equals(MediaType.OCTET_STREAM)) {
-            if (XML_MEDIA_TYPES.contains(mediaType)) {
-                return processXml(url, content);
-            } else if (TEXT_MEDIA_TYPES.contains(mediaType)) {
-                return processText(url, content);
-            } else if (GZ_MEDIA_TYPES.contains(mediaType)) {
-                InputStream decompressed;
-                MediaType embeddedType;
-                try {
-                    decompressed = new GZIPInputStream(new ByteArrayInputStream(content));
-                    embeddedType = MediaType.parse(TIKA.detect(decompressed));
-                } catch (Exception e) {
-                    UnknownFormatException err = new UnknownFormatException("Failed to detect embedded MediaType of gzipped sitemap: " + url + ", caused by " + e);
-                    err.initCause(e);
-                    throw err;
-                }
-                if (XML_MEDIA_TYPES.contains(embeddedType)) {
+        String mimeType = mimeTypeDetector.normalize(contentType, content);
+
+        if (mimeTypeDetector.isXml(mimeType)) {
+            return processXml(url, content);
+        } else if (mimeTypeDetector.isText(mimeType)) {
+            return processText(url, content);
+        } else if (mimeTypeDetector.isGzip(mimeType)) {
+            try (InputStream decompressed = new BufferedInputStream(new GZIPInputStream(new ByteArrayInputStream(content)))) {
+                String compressedType = mimeTypeDetector.detect(decompressed);
+                if (mimeTypeDetector.isXml(compressedType)) {
                     return processGzippedXML(url, content);
-                } else if (TEXT_MEDIA_TYPES.contains(embeddedType)) {
-                    // re-open decompressed stream and parse as text
-                    decompressed = new GZIPInputStream(new ByteArrayInputStream(content));
+                } else if (mimeTypeDetector.isText(compressedType)) {
                     return processText(url, decompressed);
-                } else if (GZ_MEDIA_TYPES.contains(embeddedType)) {
-                    throw new UnknownFormatException("Can't parse gzip recursively: " + url);
                 }
-                throw new UnknownFormatException("Can't parse a gzipped sitemap with the embedded MediaType of: " + embeddedType + " (at: " + url + ")");
+            } catch (Exception e) {
+                String msg = String.format("Failed to detect embedded MediaType of gzipped sitemap '%s'", url);
```

Comment:
use explicit locale?  
The forbidden API complains otherwise
```
[ERROR] Forbidden method invocation: java.lang.String#format(java.lang.String,java.lang.Object[]) [Uses default locale]
[ERROR]   in crawlercommons.sitemaps.SiteMapParser (SiteMapParser.java:232)
[ERROR] Forbidden method invocation: java.lang.String#format(java.lang.String,java.lang.Object[]) [Uses default locale]
[ERROR]   in crawlercommons.sitemaps.SiteMapParser (SiteMapParser.java:237)
```